File: 이연걸/평범한배낭.py
n, k = map(int, input().split())
stuff_list = [list(map(int, input().split())) for _ in range(n)]
dp = [0] * (k+1)

# for wei, val in stuff_list:
#     for i in range(wei, k+1):
#         dp[i] = max(val + dp[i-wei], dp[i])

for wei, val in stuff_list:
    for i in range(k, wei-1, -1):
        dp[i] = max(val + dp[i-wei], dp[i])

print(dp[k])

File: 심채윤/평범한 배낭.js
const input1 = ['4 7', '6 13', '4 8', '3 6', '5 12'] // 14
const input3 = ['3 5', '2 3', '3 2', '5 4'] // 5

const input = input3;

const [n, k]  = input[0].split(' ').map(Number)
const arr = input.slice(1).map(el => el.split(' ').map(Number))
const dp = new Array(k + 1).fill(0)

arr.sort((a, b) => a[0] - b[0]);

for (let i = 0; i < arr.length; i++) {
  const [weight, value] = arr[i]
  for (let j = k; j >= weight; j--) {
    dp[j] = Math.max(dp[j], dp[j - weight] + value);
	}
}

console.log(dp[k])